tests/one_taskset.rs

```diff
@@ -202,48 +228,69 @@ fn test_single_taskset_closure() {
         taskset = new_taskset;
     }
 
+    // Run the loadtest.
     let goose_stats = crate::GooseAttack::initialize_with_config(config.clone())
         .setup()
         .unwrap()
         .register_taskset(taskset)
         .execute()
         .unwrap();
 
-    // Confirm that we loaded the mock endpoints.
-    assert!(index.times_called() > 0);
-    assert!(about.times_called() > 0);
+    // Ensure that the right paths have been called.
+    for (idx, item) in test_endpoints.iter().enumerate() {
+        let mock_endpoint = &mock_endpoints[idx];
+
+        let format_item = |message, assert_item| {
+            return format!("{} for item = {:#?}", message, assert_item);
+        };
```

Comment:
While this could take a reference to the item in the closure, it's easier to pass it in and keep that as a pure stateless function.